activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -641,8 +658,8 @@ def load_types_queries(initializer, oids)
             yield query + "WHERE t.oid IN (%s)" % oids.join(", ")
           else
             yield query + initializer.query_conditions_for_known_type_names
-            yield query + initializer.query_conditions_for_known_type_types
             yield query + initializer.query_conditions_for_array_types
+            yield query + initializer.query_conditions_for_known_type_types
```

Comment:
`array_types` may cover `known_type_types` (ranges), which means if run `array_types` first, some `ranges` types could be skipped while invoking https://github.com/rails/rails/blob/beed933b9bb9f317697a95a36da067a67797e0d8/activerecord/lib/active_record/connection_adapters/postgresql/oid/type_map_initializer.rb#L109-L115